kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt

```diff
@@ -59,12 +60,20 @@ fun Arb.Companion.localDate(
    minDate > maxDate -> throw IllegalArgumentException("minDate must be before or equal to maxDate")
    minDate == maxDate -> Arb.constant(minDate)
    else -> {
-      val leapYears = (minDate.year..maxDate.year).filter { isLeap(it.toLong()) }
-
-      val february28s = leapYears.map { LocalDate.of(it, 2, 28) }
-      val february29s = february28s.map { it.plusDays(1) }
+      val firstPotentialLeapYear = (minDate.year + 3) and (0.inv() shl 2)
```

Comment:
This is nigh un-readable to me. Could we just do this instead?

```suggestion
      val firstPotentialLeapYear = (minDate.year..minDate.year+8).first { isLeap(it.toLong()) }
```

Reply:
My apologies, forgot to reply.
I don't mind making it more readable, but using `isLeap` makes it inconsistent.

As mentioned above, I opted for "potential leap years" due to it both being simpler and it probably even being an advantage to also have those "special cases" as edge cases for March 1. However, `isLeap` does only return true on _actual_ leap years, so we'd be mixing two different things.

However, we could just replace it by `first { it % 4 == 0 }`. Just not sure if that's really _that_ more readable; it will be a (tiny) bit slower though.

Reply:
Another thought:

```
val canDivideByFour = 0.inv() shl 2
val firstPotentialLeapYear = (minDate.year + 3) and canDivideByFour
```

Reply:
For me at least  `it % 4 == 0` clearly signals divisibility by 4. If you want to keep the bitwise arithmetic, I think your latest suggestion is better but I would also add a comment on the 2nd line explaining that you're masking away the 2 least significant bits to get a number divisible by 4, or something like it.

Reply:
Maybe calling the variable "divisibleByFourBitmask" does the trick?

`val divisibleByFourBitmask = -4` also avoids the bit shift, but the "minus" feels rather weird. However, as it's a signed int, I can't really get around that. The actual bitmask would  be written as `-0b100`, so also not overly obvious.

I'm not a big fan of using the range + `first` - it might somewhat fool-proof it, but also introduces an obscure constant (+8) that reads weird at first. Maybe really just go for:

```
val divisibleByFourBitmask = -4
val firstPotentialLeapYear = (minDate.year + 3) and divisibleByFourBitmask
```

That explains there's a bitmask involved and anyone who doesn't know enough about bitmasks at least has a starting point to look it up.

Reply:
I'm in favor of the last suggestion. Thanks for putting up with the nitpicking.. 👍🏼